source/Handlebars/Compiler/Lexer/Parsers/WordParser.cs

```diff
@@ -45,7 +45,7 @@ private string AccumulateWord(TextReader reader)
                 {
                     var peek = (char)reader.Peek();
 
-                    if (peek == '}' || peek == '~' || peek == ')' || char.IsWhiteSpace(peek))
+                    if (peek == '}' || peek == '~' || peek == ')' || (char.IsWhiteSpace(peek) && !buffer.ToString().Contains("[")))
```

Comment:
I don't think this is a safe assumption. We shouldn't merely check to see if the buffer contains a `[`, but we should check to see if the buffer contains a `[` but has no subsequent `]` to ensure we are inside the brackets.